tests/unit/Action/LoadStreamTest.php

```diff
@@ -568,6 +568,130 @@ public function it_returns_only_matched_metadata_and_properties_on_load(): void
         $this->assertSame($expected, json_decode($response->getBody()->getContents(), true));
     }
 
+    /**
+     * @test
+     */
+    public function it_converts_in_filter_value_to_array(): void
+    {
+        $uuid1 = Uuid::uuid4()->toString();
+
+        $time1 = new DateTimeImmutable('now', new DateTimeZone('UTC'));
+
+        $metadataMatcher = new MetadataMatcher();
+        $metadataMatcher = $metadataMatcher->withMetadataMatch('foo', Operator::IN(), ['bar', 'baz'], FieldType::METADATA());
+        $metadataMatcher = $metadataMatcher->withMetadataMatch('message_name', Operator::EQUALS(), 'message_one', FieldType::MESSAGE_PROPERTY());
+
+        $eventStore = $this->prophesize(EventStore::class);
+        $eventStore->load(new StreamName('foo'), 1, 3, $metadataMatcher)->willReturn(new ArrayIterator([
+            GenericEvent::fromArray([
+                'uuid' => $uuid1,
+                'message_name' => 'message_one',
+                'created_at' => $time1,
+                'payload' => ['one'],
+                'metadata' => [
+                    'foo' => 'bar',
+                ],
+            ]),
+        ]))->shouldBeCalled();
+
+        $messageConverter = new NoOpMessageConverter();
+
+        $uri = $this->prophesize(Uri::class);
+        $uri->getScheme()->willReturn('http')->shouldBeCalled();
+        $uri->getPort()->willReturn(8080)->shouldBeCalled();
+        $uri->getHost()->willReturn('localhost')->shouldBeCalled();
+
+        $request = $this->prophesize(ServerRequestInterface::class);
+        $request->getHeaderLine('Accept')->willReturn('application/vnd.eventstore.atom+json')->shouldBeCalled();
+        $request->getAttribute('streamname')->willReturn('foo')->shouldBeCalled();
+        $request->getAttribute('start')->willReturn('1')->shouldBeCalled();
+        $request->getAttribute('direction')->willReturn('forward')->shouldBeCalled();
+        $request->getAttribute('count')->willReturn('3')->shouldBeCalled();
+        $request->getUri()->willReturn($uri->reveal())->shouldBeCalled();
+        $request->getQueryParams()->willReturn([
+            'meta_0_field' => 'foo',
+            'meta_0_operator' => 'IN',
+            'meta_0_value' => 'bar;baz',
```

Comment:
Will array notation not work? Just a question

Reply:
:thinking: how should such a query look?

`?meta_0_field=_aggregate_version&meta_0_operator=IN&meta_0_value[]=2&meta_0_value[]=3`

Reply:
ok cool, this query works: `http://localhost:8081/api/v1/stream/event_stream?meta_0_field=_aggregate_version&meta_0_operator=IN&meta_0_value[]=2&meta_0_value[]=3`

Should we close the PR or still merge is a shortcut?